File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/抽象工厂模式/java/factory/Factory.java
package com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.factory;

import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.abst.ProductA;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.abst.ProductB;

public abstract class Factory {
    public abstract ProductA createProductA();
    public abstract ProductB createProductB();
}

File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/抽象工厂模式/kotlin/factory/Factory.kt
package com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.factory

import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.abst.ProductA
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.abst.ProductB

abstract class Factory {
   open abstract fun createProductA(): ProductA
   open abstract fun createProductB(): ProductB
}